qemu/tests/qemu_guest_agent.py

```diff
@@ -801,27 +801,48 @@ def gagent_check_set_time(self, test, params, env):
         delta = abs(int(guest_time_after) - target_time / 1000000000)
         if delta > 3:
             test.fail("the time set for guest is not the same with target")
-        # Set the system time from the hwclock
-        if params["os_type"] != "windows":
-            move_time_cmd = params["move_time_cmd"]
-            session.cmd("hwclock -w")
-            guest_hwclock_after_set = session.cmd_output("date +%s")
-            error_context.context("hwclock is '%d' " % int(guest_hwclock_after_set),
-                                  logging.info)
-            session.cmd(move_time_cmd)
-            time_after_move = session.cmd_output("date +%s")
-            error_context.context("the time after move back is '%d' "
-                                  % int(time_after_move), logging.info)
-            self.gagent.set_time()
-            guest_time_after_reset = session.cmd_output(get_guest_time_cmd)
-            error_context.context("the time after being reset is '%d' "
-                                  % int(guest_time_after_reset), logging.info)
-            guest_hwclock = session.cmd_output("date +%s")
-            error_context.context("hwclock for compare is '%d' " % int(guest_hwclock),
+
+        # set invalid dometime if needed
+        invalid_time_test = params.get_boolean("invalid_time_test", "no")
+        if invalid_time_test:
+            error_context.context("Set time to an invalid value.",
                                   logging.info)
-            delta = abs(int(guest_time_after_reset) - int(guest_hwclock))
-            if delta > 3:
-                test.fail("The guest time can't be set from hwclock on host")
+            guest_time_before = session.cmd_output(get_guest_time_cmd)
+            target_time_invalid = int(guest_time_before) * 1000000000000
+            try:
+                self.gagent.set_time(target_time_invalid)
+            except guest_agent.VAgentCmdError as e:
+                expected = "Invalid parameter type"
+                if expected not in e.edata["desc"]:
+                    test.fail(e)
+            delta = abs(int(guest_time_after) - int(guest_time_before))
```

Comment:
`guest_time_after` is from Line 798? So which one is 'before' and which one is 'after', and what do you want to compare? Looks confusing here.